src/vivarium_gates_nutrition_optimization_child/components/lbwsg.py

```diff
@@ -206,3 +232,63 @@ def get_birth_exposure(self, axis: str, index: pd.Index) -> pd.DataFrame:
             exposure_values.loc[subset_index] = lbwsg_exposures[axis].values
 
         return exposure_values
+
+
+class LBWSGPAFObserver(Component):
+    CONFIGURATION_DEFAULTS = {
+        "stratification": {
+            "lbwsg_paf": {
+                "exclude": [],
+                "include": [],
+            }
+        }
+    }
+
+    @property
+    def columns_required(self) -> Optional[List[str]]:
+        return ["lbwsg_category"]
+
+    def __init__(self, target: str):
+        super().__init__()
+        self.target = TargetString(target)
+
+    # noinspection PyAttributeOutsideInit
+    def setup(self, builder: Builder) -> None:
+        self.lbwsg_exposure = builder.data.load(data_keys.LBWSG.EXPOSURE)
+        self.risk_effect = builder.components.get_component(
+            f"lbwsgpaf_calculation_risk_effect.{self.target}"
+        )
+        self.config = builder.configuration.stratification.lbwsg_paf
+
+        builder.results.register_observation(
+            name=f"calculated_lbwsg_paf_on_{self.target}",
+            pop_filter='alive == "alive"',
+            aggregator=self.calculate_paf,
+            requires_columns=["alive"],
+            additional_stratifications=self.config.include,
+            excluded_stratifications=self.config.exclude,
+            when="time_step__prepare",
```

Comment:
I don't have a strong sense of what you're doing here, but can you confirm this is the correct spot to run?

Reply:
This was copied from the CVD paf calculation. I can confirm that everything that we need has been initialized here (and that PAFs passed V&V from research :) )